scipy/spatial/transform/_rotation.pyx

```diff
@@ -1865,6 +1873,118 @@ cdef class Rotation:
             result = result[0]
         return self.__class__(result, normalize=True, copy=False)
 
+    @cython.embedsignature(True)
+    def __pow__(Rotation self, double n, modulus):
+        """Compose this rotation with itself ``n`` times.
+
+        Composition of a rotation `p` with itself can be extended to
+        non-integer ``n`` by considering the power ``n`` to be a scale factor
+        applied to the angle of rotation about the rotation's fixed axis. The
+        expression ``q = p ** n`` can also be expressed as
+        ``q = Rotation.from_rotvec(n * p.as_rotvec())``
+
+        Parameters
+        ----------
+        n : `double`
+            The number of times to compose the rotation with itself. ``n`` must
+            be positive.
+        modulus : None
+            This overridden argument is not applicable to Rotations and must be
+            None.
+
+        Returns
+        -------
+        composition : `Rotation` instance
+            If the input Rotation ``p`` contains ``N`` multiple rotations, then
+            the output will contain ``N`` rotations where the ``i``th rotation
+            is equal to ``p[i] ** n``
+
+        Notes
+        -----
+        For example, a power of 2 will double the angle of rotation, and a
+        power of 0.5 will halve the angle. There are two notable cases: if
+        ``n == 1`` then the original rotation is returned, and if ``n == 0``
+        then the identity rotation is returned.
+
+        Note that fractional powers ``0 < n < 1`` which effectively take a root
+        of rotation, do so using the shortest path smallest representation of
+        that angle (the principal root). This means that powers of ``n`` and
+        ``1/n`` are not necessarily inverses of each other. For example, a 0.5
+        power of a +240 degree rotation will be calculated as the 0.5 power of
+        a -120 degree rotation, with the result being a rotation of -60 rather
+        than +120 degrees.
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import Rotation as R
+
+        Raising a rotation to a power:
+
+        >>> p = R.from_rotvec([1, 0, 0])
+        >>> q = p ** 2
+        >>> q.as_rotvec()
+        array([2., 0., 0.])
+        >>> r = p ** 0.5
+        >>> r.as_rotvec()
+        array([0.5, 0., 0.])
+
+        Inverse powers do not necessarily cancel out:
+
+        >>> p = R.from_rotvec([0, 0, 120], degrees=True)
+        >>> ((p ** 2) ** 0.5).as_rotvec(degrees=True)
+        array([  -0.,   -0., -60.])
+        """
+        cdef Py_ssize_t num_rotations = len(self._quat)
+        cdef double angle, scale, angle2, n3, n5
+        cdef double[:, :] result = _empty2(num_rotations, 4)
+        cdef double[:] quat
+
+        if modulus is not None:
+            raise NotImplementedError("modulus not supported")
+
+        if n < 0:
+            raise ValueError("Rotations cannot be raised to negative powers.")
```

Comment:
Maybe a silly question, but why not implement negative powers as the inverse rotation?
Since, for example `q**(-2) = (q**2)**(-1)`

Reply:
I simply hadn’t seen that notation anywhere before, but it makes sense to me and as you say should be easy to add.

Reply:
This would imply to me that we should also implement the `__div__` operator, such that the following are equivalent: `A / B == A * B.inv()`.
@rkern have you see that notation anywhere?

Reply:
Less so because of non-commutativity. It's easier to express the right things with `*` and negative powers.

Reply:
Added support for negative exponents.